pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -286,8 +287,17 @@ func (r *Runtime) runGadget(gadgetCtx runtime.GadgetContext, pod gadgetPod, allP
 	jsonArrayHandler := func([]byte) {}
 
 	if parser != nil {
-		jsonHandler = parser.JSONHandlerFunc()
-		jsonArrayHandler = parser.JSONHandlerFuncArray(pod.node)
+		var enrichers []func(any) error
+		ev := gadgetCtx.GadgetDesc().EventPrototype()
+		if _, ok := ev.(operators.NodeSetter); ok {
+			enrichers = append(enrichers, func(ev any) error {
+				ev.(operators.NodeSetter).SetNode(pod.node)
+				return nil
+			})
+		}
```

Comment:
This enrichment will only be needed by gadgets that doesn't use a parser?

Reply:
I think so, gadgets that don't use a parser only return `[]bytes` and we don't have any way to enrich it.